### PR TITLE
WIP: Updated documentation for the upcoming website update

### PR DESCRIPTION
**Pull request by dragomirt**
This PR contains changes in the documentation files. It removes redundant metadata from the markdown headers, in this case, it's `categories`, adds an h1 header at the top of the doc for a proper search indexation and modifies the file names. The images were as well moved into the docs folder for MkDocs compatibility. The contents of the `index.md` file are temporary.

### CODE REVIEW

File: website/docs/_posts/assets/js/fx.js [hunk 1]
```diff
@@ -0,0 +1 @@
+document.title += " | Renovate";
```
Comment (rarkins) [inline website/docs/_posts/assets/js/fx.js]: Website assets shouldn't be stored in this repository. Is there no other way to get the suffix using mkdocs natively?

File: website/docs/_posts/assets/css/style.css [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.md-header-nav__button.md-logo *{
+	height: 35px;
+	width: 35px;
+	margin-top: -5px;
+	margin-right: -10px;
+}
```
Comment (rarkins) [inline website/docs/_posts/assets/css/style.css]: move this into the repo for renovatebot.com

File: website/docs/_posts/configuration-options.md [hunk 3]
```diff
@@ -908,15 +908,6 @@ Whether to convert ranged versions in `package.json` to pinned versions.
 
 This is a very important feature to consider, because not every repository's requirements are the same. The default value within the tool itself is false, which means no existing ranges are pinned. However if you are using the suggested preset `"config:base"`, then it changes the default of pinVersions to `null`, which means Renovate attempts to autodetect what's best for the project. In such cases `devDependencies` in `package.json` will alway be pinned, but `dependencies` will only be pinned if the package is `private` or has no `main` entry defined - both indicators that it is not intended to be published and consumed by other packages. To override the `"config:base"` setting, add the preset `":preserveSemverRanges"` to your `extends` array.
 
-## pip_requirements
```
Comment (rarkins) [inline website/docs/_posts/configuration-options.md]: why is this removed?

File: website/docs/_posts/configuration-options.md [hunk 4]
```diff
@@ -996,15 +987,6 @@ Pull Request title template
 
 The PR title is important for some of Renovate's matching algorithms (e.g. determining whether to recreate a PR or not) so ideally don't modify it much.
 
-## python
```
Comment (rarkins) [inline website/docs/_posts/configuration-options.md]: why is this removed?

File: website/docs/_posts/configuration-options.md [hunk 5]
```diff
@@ -1204,7 +1186,7 @@ Dependency support policy, e.g. used for deciding whether to use a [Long-term Su
 
 Language support is limited to those listed below:
 
-* **Node.js** - [Read our Node.js documentation](https://renovateapp.com/docs/language-support/node#configuring-support-policy)
+* **Node.js** - [Read our Node.js documentation](https://renovatebot.com/docs/node/)
```
Comment (rarkins) [inline website/docs/_posts/configuration-options.md]: Can we have them like docs/language/node ?

File: website/docs/_posts/index.md [hunk 6]
```diff
@@ -0,0 +1 @@
+#hi
```
Comment (rarkins) [inline website/docs/_posts/index.md]: This can be moved to the renovatebot.com repository

File: website/docs/_posts/configuration-options.md (hunk 3 above)
Comment (SevenTurtles) [reply website/docs/_posts/configuration-options.md]: It will be added back again after merging with master.

Comment (rarkins) [reply website/docs/_posts/configuration-options.md]: But is this just because your branch/fork is out of date with the `master` of this repo? I much prefer you update your fork than to merge and have it out of date immediately

File: website/docs/_posts/configuration-options.md [hunk 7]
```diff
@@ -172,11 +172,11 @@ Set this value to 'patch', 'minor' or 'major' to have Renovate update the versio
...
-| name     | value  |
-| -------- | ------ |
-| type     | object |
-| default  | {}     |
-| language | docker |
+| name          | value  |
+| ------------- | ------ |
+| type          | object |
+| default       | {}     |
+| parentManager | docker |
```
Comment (rarkins) [inline website/docs/_posts/configuration-options.md]: No, this is a bad revert

File: website/docs/_posts/configuration-options.md [hunk 8]
```diff
@@ -272,6 +272,17 @@ A list of depType names inside a package rule to filter on. Matches all depTypes
 
 Use this field if you want to limit a `packageRule` to certain `depType` values. Invalid if used outside of a `packageRule`.
 
+## dependencies
```
Comment (rarkins) [inline website/docs/_posts/configuration-options.md]: No

File: website/docs/_posts/configuration-options.md [hunk 9]
```diff
@@ -281,6 +292,17 @@ Use this field if you want to limit a `packageRule` to certain `depType` values.
 
 The description field is used by config presets to describe what they do. They are then collated as part of the onboarding description.
 
+## devDependencies
```
Comment (rarkins) [inline website/docs/_posts/configuration-options.md]: No

File: website/docs/_posts/configuration-options.md [hunk 10]
```diff
@@ -305,11 +327,11 @@ Configuration specific for Dockerfile updates.
...
-| name     | value  |
-| -------- | ------ |
-| type     | object |
-| default  | {}     |
-| language | docker |
+| name          | value  |
+| ------------- | ------ |
+| type          | object |
+| default       | {}     |
+| parentManager | docker |
```
Comment (rarkins) [inline website/docs/_posts/configuration-options.md]: No

File: website/docs/_posts/configuration-options.md [hunk 11]
```diff
@@ -439,15 +461,6 @@ Preset configs to use/extend.
 
 See https://renovateapp.com/docs/configuration-reference/config-presets for details.
 
-## fileMatch
```
Comment (rarkins) [inline website/docs/_posts/configuration-options.md]: No

File: website/docs/_posts/configuration-options.md [hunk 12]
```diff
@@ -547,19 +560,6 @@ Ignore versions with unstable semver.
 
 By default, Renovate won't update any packages to unstable versions (e.g. `4.0.0-rc3`) unless the package version was _already_ unstable (e.g. it was already on `4.0.0-rc2`). If for some reason you wish to _force_ unstable updates on Renovate, you can set `ignoreUnstable` to `false`, but this is not recommended for most situations.
 
-## includePaths
```
Comment (rarkins) [inline website/docs/_posts/configuration-options.md]: No

File: website/docs/_posts/configuration-options.md [hunk 13]
```diff
@@ -716,14 +716,36 @@ See https://renovateapp.com/docs/deep-dives/private-modules for details on how t
...
-| name     | value  |
-| -------- | ------ |
-| type     | object |
-| default  | { }    |
-| language | node   |
+| name          | value  |
+| ------------- | ------ |
+| type          | object |
+| default       | { }    |
+| parentManager | node   |
```
Comment (rarkins) [inline website/docs/_posts/configuration-options.md]: No

**Comment by rarkins** [comment]
@SevenTurtles there is a massive amount of bad changes in the configuration options doc now. Almost all of them need reverting/fixing. You may be better copy/pasting the doc from master and reapplying your own changes

**Comment by rarkins** [comment]
Also, we should reconsider the file paths in use now, if we use mkdocs instead of jekyll to generate renovatebot.com/docs/. i.e. instead of `website/docs/_posts/bazel.md` I think it should probably be something like `website/docs/managers/bazel.md` instead.

**Comment by rarkins** [comment]
Another point: if the jekyll front matter is no longer used, then we should remove it, e.g.

```
date: 2017-07-31
title: Private npm module support
description: How to support private npm  modules when using Renovate
type: Document
order: 30
```

Are date, description, type and order used - or just title?

**Comment by rarkins** [comment]
I made the configuration options doc reverting/fixing

**Comment by SevenTurtles** [comment]
I left the description for any eventualities, we might use it later.
